### DISCUSSION TITLE
Is there any way to input custom masterKey during Gocryptfs initialization?

### DISCUSSION THREAD

**Discussion by anjalig6** [rfjakob/gocryptfs#774] [General]
I have a use-case where I want to input custom masterKey during Gocryptfs initialization and not use the auto-generated one. Is that doable?

**Comment by rfjakob** [comment]
I think (untested) you can do this:

1) Do normal `gocryptfs -init`
2) DO NOT ADD ANY DATA YET
3) Do `gocryptfs -passwd -masterkey=YOUR-MASTER-KEY` to change the masterkey to what you want
